### DISCUSSION TITLE
Why the allocator uses an instance of Object instead of Free

### DISCUSSION THREAD

**Discussion by lordfanger** [kevingosse/kevingosse.github.io#2] [General]
Hi, I couldn't find another way to contact you. (And I don't want to give rights to [utteranc.es](https://utteranc.es/))
I've found your great article https://minidump.net/memory-alignment-of-doubles-in-c-1d13e3ce741/ and my idea why Object is prefered over Free is that there can be some bookkeeping (I found some `freelist` in the docs) or that there could be two Free objects next to each other. If there were already Free object before current pointer (with aligned access at slow path for example) and in docs says these should be merged and that Free objects should be created at sweep phase.
https://github.com/dotnet/runtime/blob/f2a9ef8d392b72e6f039ec0b87f3eae4307c6cae/docs/design/coreclr/botr/garbage-collection.md#sweep-phase

But I'm not expert, this is only my idea.

**Comment by kevingosse** [comment]
You're right that's the free-list could be part of the answer 🤔  Unfortunately there's no way to be sure. Thanks for the feedback!
